drivers/stepper/allegro/a4979.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT allegro_a4979
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/drivers/gpio.h>
+#include "../step_dir/step_dir_stepper_common.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(a4979, CONFIG_STEPPER_LOG_LEVEL);
+
+struct a4979_config {
+	const struct step_dir_stepper_common_config common;
+	const struct gpio_dt_spec en_pin;
+	const struct gpio_dt_spec reset_pin;
+	const struct gpio_dt_spec m0_pin;
+	const struct gpio_dt_spec m1_pin;
+};
+
+struct a4979_data {
+	const struct step_dir_stepper_common_data common;
+	enum stepper_micro_step_resolution micro_step_res;
+	bool enabled;
+};
+
+STEP_DIR_STEPPER_STRUCT_CHECK(struct a4979_config, struct a4979_data);
+
+static int a4979_set_microstep_pin(const struct device *dev, const struct gpio_dt_spec *pin,
+				   int value)
+{
+	int ret;
+
+	/* Reset microstep pin as it may have been disconnected. */
+	ret = gpio_pin_configure_dt(pin, GPIO_OUTPUT_INACTIVE);
+	if (ret != 0) {
+		LOG_ERR("Failed to configure microstep pin (error: %d)", ret);
+		return ret;
+	}
+
+	ret = gpio_pin_set_dt(pin, value);
+	if (ret != 0) {
+		LOG_ERR("Failed to set microstep pin (error: %d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int a4979_stepper_enable(const struct device *dev, bool enable)
+{
+	int ret;
+	const struct a4979_config *config = dev->config;
+	struct a4979_data *data = dev->data;
+	bool has_enable_pin = config->en_pin.port != NULL;
+
+	/* Check availability of enable pin, as it might be hardwired. */
+	if (has_enable_pin) {
```

Comment:
invert the condition, return the error in the branch and keep the normal exection unindented